app/models/components/course/materials_ability_component.rb

```diff
@@ -63,4 +63,22 @@ def concrete_folder_hash
     # Linked folders(folders with owners) are not manageable
     { owner_id: nil }
   end
+
+  # Involve Course#advance_start_at_duration when calculating the start_at time.
+  def opened_material_hashes
+    max_start_at = Time.zone.now
+    # Extend start_at time with self directed time from course settings.
+    max_start_at += (course.advance_start_at_duration || 0) if course
```

Comment:
Would it be good to have this extracted as a helper function for all course components?